### PR TITLE
fix(helmv3): adapt to new OCI format and allow aliases for OCI

### LINKED ISSUE #12199: Helmv3 OCI dependencys with aliases
### What would you like Renovate to be able to do?

OCI registry lookup with the helmv3 manager should also work with aliases.

renovate.json:
```json
{
  "helmv3": {
    "aliases": {
      "artifactory": "oci://artifactory.example.com"
    }
  }
}
```

Chart.yaml:
```yaml
apiVersion: v2
description: Renovate example
name: example
version: 0.0.1
dependencies:
  - name: "example"
    version: "1.0.2"
    repository: "alias:artifactory"
```

### If you have any ideas on how this should be implemented, please tell us here.

Remove/rewrite [this](https://github.com/renovatebot/renovate/blob/c43554b69343fb131209997c52cac518b3048fe0/lib/manager/helmv3/extract.ts#L60) if/else block

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by secustor**
## Changes:
- Adapts to the new OCI helm chart format.      
  Chart name is now appended to repository    https://helm.sh/docs/topics/registries/#deprecated-features-and-strict-naming-policies        
This is potentially breaking to user  of Helm < 3.7 and OCI registries.     
- Ensure that aliases of OCI registries are resolved 
- small refactor changes to simplify logic and improve readability 

Not sure if we should handle this as breaking change, as it is an experimental feature of Helm.      

## Context:
Closes #12199
Fixes the schema change referenced in #13524 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository
https://github.com/secustor/renovate-helm-oci

### CODE REVIEW

**Comment by rarkins** [comment]
How could "user of Helm < 3.7 and OCI registries" unbreak themselves after this change?

File: lib/manager/helmv3/utils.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  return res;
+}
+
+// Resolves alias in repository string.
+// If repository does not contain an alias, then the repository string will be returned.
+// Should the resolution fail, `null` will be returned .
+export function resolveAlias(
+  repository: string,
+  aliases: Record<string, string>
+): string {
```
Comment (viceice) [inline lib/manager/helmv3/utils.ts:38]:
```suggestion
): string | null {
```

**Review by viceice — COMMENTED** [review]
Normally it would better to have the refactor first, so we have less changes.

**Comment by secustor** [comment]
> In addition, a change to the experimental OCI manifest format was introduced, preventing Helm 3.7 from fetching charts built
with older versions of the experimental OCI feature. Users will need to re-package and upload their OCI charts when upgrading to Helm 3.7. This change does not affect standard Helm charts.
> Many of the details of these changes can be found in HIP 6.

https://github.com/helm/helm/releases/tag/v3.7.0

Users have to republish their charts.

@viceice I can break out the schema change as it is only two lines. The refactoring simply has fallen in place during debugging and is pretty straight forward, so I have chosen to propose it together.

**Comment by rarkins** [comment]
Is there something fundamentally stopping *us* supporting both though?

**Comment by secustor** [comment]
K, I have tested this with Helm 3.5, 3.6 and 3.7.      
This is actually a bug on our side and not a change of Helm.      

All but one change is referring to file structure inside of the OCI images or CLI changes.    
> The chart layer media type was switched from application/tar+gzip to application/vnd.cncf.helm.chart.content.v1.tar+gzip

We could use this to theoretically detect if this is pre 3.7, but from the users perspective this brings IMHO no value. If some user wants to keep using the old format they simply have to keep pushing their charts with Helm 3.6 or earlier.     

Therefore I would downgrade that to a simple fix.

**Review by rarkins — DISMISSED** [review]

File: lib/manager/helmv3/utils.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        res.registryUrls = [repositoryURL];
+    }
+  } catch (err) {
+    logger.debug({ err }, 'Error parsing url');
+    res.skipReason = SkipReason.InvalidUrl;
+  }
+  return res;
+}
+
+// Resolves alias in repository string.
```
Comment (viceice) [inline lib/manager/helmv3/utils.ts:32]: Please use tsdoc/ jsdoc format.

**Review by viceice — COMMENTED** [review]
Just a small thing left.

**Review by viceice — APPROVED** [review]

**Pull request merged**